### PR TITLE
Add support for ValueTuple

### PR DESCRIPTION
**Pull request by rodriguestiago0**

### CODE REVIEW

File: src/FSharpPlus/FSharpPlus.fsproj [hunk 1]
```diff
@@ -51,6 +51,7 @@
     <Compile Include="Extensions/Async.fs" />
     <Compile Include="Extensions/Extensions.fs" />
     <Compile Include="Extensions/Tuple.fs" />
+    <Compile Include="Extensions\ValueTuple.fs" />
```
Comment (gusty) [inline src/FSharpPlus/FSharpPlus.fsproj:54]: Bar `\`

File: src/FSharpPlus/Operators.fs [hunk 2]
```diff
@@ -114,8 +115,48 @@ module Operators =
...
+    /// <category index="0">Common Combinators</category>
+    let inline valueTuple4<'T1, 'T2, 'T3, 'T4> (t1: 'T1) (t2: 'T2) (t3: 'T3) (t4: 'T4) = ValueTuple.Create t1, t2, t3, t4
+
+    /// <summary>
+    /// ValueTuple five arguments
+    /// </summary>
+    /// <category index="0">Common Combinators</category>
+    let inline valueTuple5<'T1, 'T2, 'T3, 'T4, 'T5> (t1: 'T1) (t2: 'T2) (t3: 'T3) (t4: 'T4) (t5: 'T5) = ValueTuple.Create t1, t2, t3, t4, t5
 
+    /// Tuple six arguments
```
Comment (gusty) [inline src/FSharpPlus/Operators.fs:143]: `ValueTuple`

File: src/FSharpPlus/FSharpPlus.fsproj (hunk 1 above)
Comment (rodriguestiago0) [inline src/FSharpPlus/FSharpPlus.fsproj:54]:
```suggestion
    <Compile Include="Extensions/ValueTuple.fs" />
```

File: src/FSharpPlus/Operators.fs (hunk 2 above)
Comment (rodriguestiago0) [inline src/FSharpPlus/Operators.fs:143]:
```suggestion
    /// ValueTuple six arguments
```

File: docsrc/content/abstraction-semigroup.fsx [hunk 3]
```diff
@@ -42,6 +42,7 @@ From .Net/F#
  -  ``Map<'T,'U>``
  -  ``TimeSpan`` 
  -  ``Tuple<*>``
+ -  ``ValueTuple<*>``
```
Comment (gusty) [inline docsrc/content/abstraction-semigroup.fsx:45]: This is tricky. either we support `ValueTuple<n>` with `n < 8` or we do some kind of reflection trick for those tuples greater than 8.

Comment (rodriguestiago0) [reply docsrc/content/abstraction-semigroup.fsx:45]: we support for n > 0 and n <9

Comment (rodriguestiago0) [reply docsrc/content/abstraction-semigroup.fsx:45]: Should I remove this line?

Comment (gusty) [reply docsrc/content/abstraction-semigroup.fsx:45]: No, in that case just change it to `ValueTuple<*> (* up to 8 elements)`

**Pull request merged**